src/main/java/Parser.java

```diff
@@ -0,0 +1,71 @@
+import java.util.Scanner;
+
+/**
+ *  makes sense of the user command
+ */
+public class Parser {
+
+    Scanner scan;
+    Ui ui = new Ui();
+
+    public Parser() {
+        this.scan = new Scanner(System.in);
+    }
+
+    /**
+     * scanner scan task's instruction and run accordingly
+     * @throws DukeException when users input task which bot does not understand
+     */
+    public void scan() throws DukeException {
+        ui.greeting();
+        String command = scan.nextLine();
+        while (!command.equals("bye")) {
+            if (command.contains("find")) {
+                TaskList task = new TaskList(command);
+                task.find();
+                command = scan.nextLine();
+
+            } else if (command.contains("delete")) {
+                TaskList task = new TaskList(command);
+                task.delete();
+                command = scan.nextLine();
+
+            } else if (!command.contains("todo")
+                    && !command.contains("deadline")
+                    && !command.contains("event")
+                    && !command.contains("done")
+                    && !command.contains("list")
+                    && !command.contains("delete")) {
+                throw new DukeException(" ))-: OOPS!!! I'm sorry, "
+                        + "but I don't know what that means :-(");
```

Comment:
Consider reducing this to simpler terms with the use of `default` in switch case or just a simple `else`